registration-processor/pre-processor/registration-processor-quality-classifier-stage/src/test/java/io/mosip/registration/processor/quality/qualifier/stage/test/QualityClassifierStageTest.java

```diff
@@ -381,54 +415,54 @@ public void testQualityCheckfailureException() throws BiometricException {
 		Mockito.when(bioApiFactory.getBioProvider(any(), any())).thenReturn(null);
 		MessageDTO dto = new MessageDTO();
 		dto.setRid("1234567890");
-		MessageDTO result = qualityCheckerStage.process(dto);
+		MessageDTO result = qualityClassifierStage.process(dto);
 
 		assertTrue(result.getInternalError());
 	}
 
 	@Test
 	public void testFileMissing() throws ApisResourceAccessException, IOException, PacketManagerException, JsonProcessingException {
 		Mockito.when(registrationStatusService.getRegistrationStatus(any())).thenReturn(registrationStatusDto);
-		when(packetManagerService.getBiometricsByMappingJsonKey(anyString(),any(), any(),any())).thenReturn(null);
+		when(basedPacketManagerService.getBiometricsByMappingJsonKey(anyString(),any(), any(),any())).thenReturn(null);
 
 		MessageDTO dto = new MessageDTO();
 		dto.setRid("1234567890");
-		MessageDTO result = qualityCheckerStage.process(dto);
+		MessageDTO result = qualityClassifierStage.process(dto);
 
 		assertTrue(result.getInternalError());
 	}
 
 	@Test
 	public void testBioetricFileNotPresentInIdObject() throws ApisResourceAccessException, IOException, PacketManagerException, JsonProcessingException {
 
-		when(packetManagerService.getFieldByMappingJsonKey(any(), any(), any(), any())).thenReturn(null);
+		when(basedPacketManagerService.getFieldByMappingJsonKey(any(), any(), any(), any())).thenReturn(null);
 
 		MessageDTO dto = new MessageDTO();
 		dto.setRid("1234567890");
-		MessageDTO result = qualityCheckerStage.process(dto);
+		MessageDTO result = qualityClassifierStage.process(dto);
 
 		assertTrue(result.getIsValid());
 	}
 
 	@Test
 	public void testNoBiometricInPacket() throws ApisResourceAccessException, IOException, PacketManagerException, JsonProcessingException {
 
-		when(packetManagerService.getBiometricsByMappingJsonKey(any(),any(),any(),any())).thenReturn(null);
+		when(basedPacketManagerService.getBiometricsByMappingJsonKey(any(),any(),any(),any())).thenReturn(null);
 
 		MessageDTO dto = new MessageDTO();
 		dto.setRid("1234567890");
-		MessageDTO result = qualityCheckerStage.process(dto);
+		MessageDTO result = qualityClassifierStage.process(dto);
 
 		assertFalse(result.getIsValid());
 	}
 
 
 	@Test
 	public void testJsonProcessingException() throws ApisResourceAccessException, IOException, PacketManagerException, JsonProcessingException {

```

Comment:
You have to add unit test cases to check if the tagging method is called with proper tag created based on the score. Have 3 test cases one with 
1. All bio types are returning same (Good for all 3)
2. returns 2 combinations, (1 Good and 2 Poor), 
3. All 3 bio types are giving 3 different classifications (1 Good, 1 Poor, 1 Average), lets check if the tags are created properly based on the input score

https://github.com/mosip/registration/blob/develop/registration-processor/pre-processor/registration-processor-packet-classifier-stage/src/test/java/io/mosip/registration/processor/stages/packetclassifier/tagging/impl/AgeGroupTagGeneratorTest.java

Refer below for ArgumentCaptor to get the tags that is passed by QualityCheckerStage to addOrUpdate tags method
https://github.com/mosip/registration/blob/develop/registration-processor/registration-processor-core/src/test/java/io/mosip/registration/processor/core/eventbus/KafkaMosipEventBusTest.java

Reply:
Added UT for POOR, AVG,GOOD tags, but couldnt write it for mix and match